src/Nox.Solution/Resolvers/NoxYamlSerializer.cs

```diff
@@ -0,0 +1,137 @@
+        var yamlObjectInstance = deserializer.Deserialize<object>(yamlContent);
+        var errorsFromObjectValidation = Validate(yamlObjectInstance, jsonSerializerOptions, schema, evaluateOptions);
+        errors.AddRange(errorsFromObjectValidation);
+
+        T yamlTypedObjectInstance;
+        try
+        {
+            // Then when all required fields are validate it's neccessary to check whether all properties match T object properties.
+            // In case some field is missed it will throw an exception regarding to deserializtion error.
+            yamlTypedObjectInstance = deserializer.Deserialize<T>(yamlContent);
+        }
+        catch (YamlException ex)
+        {
+            HandleYamlExceptionMessage(ex, errors);
+            var message = string.Join("\n", errors.Distinct());
+
+            throw new NoxSolutionConfigurationException(message, ex);
+        }
+
+        var errorsFromTypedObjectValidation = Validate(yamlTypedObjectInstance, jsonSerializerOptions, schema, evaluateOptions);
+        errors.AddRange(errorsFromTypedObjectValidation);
+
+        if (errors.Count > 0)
+        {
+            throw new NoxSolutionConfigurationException(string.Join("\n", errors.Distinct()));
+        }
+
+        return yamlTypedObjectInstance;
+    }
+
+    private static void HandleYamlExceptionMessage(Exception? exception, List<string> errors)
+    {
+        if (exception is null)
+        {
+            return;
+        }
+
+        string message;
+        if (exception is YamlException yamlException)
+        {
+            message = $"{yamlException.Message}. Line {yamlException.End.Line}, column {yamlException.End.Column}.";
+        }
+        else
+        {
+            message = exception.Message;
+        }
+
+        errors.Add(message);
+        HandleYamlExceptionMessage(exception.InnerException, errors);
+    }
+
+    private static List<string> Validate<T>(
+            T yamlObject,
+            JsonSerializerOptions jsonSerializerOptions,
+            JsonSchema schema,
+            EvaluationOptions evaluateOptions)
+    {
+        var jsonDocument = JsonSerializer.SerializeToDocument(yamlObject, jsonSerializerOptions);
+        var errors = new List<string>();
+        var result = schema.Evaluate(jsonDocument, evaluateOptions);
+
+        HandleErrorsRecursively(result, errors);
+
+        return errors;
+    }
+
+    private static void HandleErrorsRecursively(EvaluationResults results, List<string> errors)
+    {
+        if (results.Errors != null)
+        {
+            foreach (var error in results.Errors)
+            {
+                if (results.EvaluationPath.ToString().EndsWith("/$ref")
+                    || error.Key == "type"
+                    )
+                {
+                    continue;
+                }
+                var path = string.IsNullOrEmpty(results.EvaluationPath.ToString()) ? string.Empty : $"Path: {results.EvaluationPath}. ";
```

Comment:
you can avoid an extra string creation:
```suggestion
             string evaluationPath = results.EvaluationPath.ToString();
             var path = string.IsNullOrEmpty(evaluationPath) ? string.Empty : $"Path: {evaluationPath}. ";
```